src/Microsoft.AspNet.Mvc.Core/ModelBinding/CollectionModelBinder.cs

```diff
@@ -56,31 +56,13 @@ public class CollectionModelBinder<TElement> : ICollectionModelBinder
             var valueProviderResult = await bindingContext.ValueProvider.GetValueAsync(bindingContext.ModelName);
 
             CollectionResult result;
-            if (valueProviderResult == null)
+            if (valueProviderResult == ValueProviderResult.None)
             {
                 result = await BindComplexCollection(bindingContext);
             }
             else
             {
-                if (valueProviderResult.RawValue == null)
-                {
-                    // Value exists but is null. Handle similarly to fallback case above. This avoids a
-                    // ModelBindingResult with IsModelSet = true but ValidationNode = null.
-                    model = bindingContext.Model ?? CreateEmptyCollection(bindingContext.ModelType);
-                    var validationNode =
-                        new ModelValidationNode(bindingContext.ModelName, bindingContext.ModelMetadata, model);
-
-                    return new ModelBindingResult(
-                        model,
-                        bindingContext.ModelName,
-                        isModelSet: true,
-                        validationNode: validationNode);
-                }
```

Comment:
There's no such thing as a `ValueProviderResult` with no value other than `ValueProviderResult.None`. Value providers have to plug in the empty string to return a "key was present but didn't have a value".

This case will now just go down the normal simple collection path. It was a nice surprise that it wasn't needed anymore.